pubtools/_quay/operator_pusher.py

```diff
@@ -421,30 +421,60 @@ def build_index_images(self):
         iib_results = {}
 
         for version, items in sorted(self.version_items_mapping.items()):
-            bundles = [self.public_bundle_ref(i) for i in items]
-            all_archs = [
-                i.metadata["arch"] if i.metadata["arch"] != "x86_64" else "amd64" for i in items
-            ]
-            archs = sorted(list(set(all_archs)))
-            signing_keys = sorted(list(set([item.claims_signing_key for item in items])))
+            is_hotfix = any([item.metadata.get("com.redhat.hotfix") for item in items])
+            is_advisory_source = all(
+                [re.match(r"^[A-Z0-9:\-]{4,40}$", item.origin) for item in items]
+            )
+            item_groups = {}
+            if is_hotfix and not is_advisory_source:
+                raise ValueError("Cannot push hotfixes without an advisory")
+            if is_hotfix:
+                for item in items:
+                    item_groups.setdefault(item.origin, []).append(item)
+            else:
+                item_groups["default"] = items
 
             # Get deprecation list
             deprecation_list = self.get_deprecation_list(version)
 
-            # build index image in IIB
-            index_image = "{image_repo}:{tag}".format(
-                image_repo=self.target_settings["iib_index_image"], tag=version
-            )
-            build_details = self.iib_add_bundles(
-                bundles=bundles,
-                archs=archs,
-                index_image=index_image,
-                deprecation_list=deprecation_list,
-                build_tags=["{0}-{1}".format(index_image.split(":")[1], self.task_id)],
-                target_settings=self.target_settings,
-            )
+            for group, g_items in item_groups.items():
+                if is_hotfix:
+                    tag = "{0}-{1}-{2}".format(
+                        version,
+                        g_items[0].metadata["com.redhat.hotfix"],
+                        g_items[0].origin.replace(":", "-"),
+                    )
+                else:
+                    tag = version
+                index_image = "{image_repo}:{tag}".format(
+                    image_repo=self.target_settings["iib_index_image"], tag=tag
```

Comment:
--index-image is used as from_index in an IIB build, it should be image with version tag.